static/app/views/settings/account/apiApplications/details.tsx

```diff
@@ -10,131 +9,155 @@ import Confirm from 'sentry/components/confirm';
 import Form from 'sentry/components/forms/form';
 import FormField from 'sentry/components/forms/formField';
 import JsonForm from 'sentry/components/forms/jsonForm';
+import LoadingError from 'sentry/components/loadingError';
+import LoadingIndicator from 'sentry/components/loadingIndicator';
 import Panel from 'sentry/components/panels/panel';
 import PanelBody from 'sentry/components/panels/panelBody';
 import PanelHeader from 'sentry/components/panels/panelHeader';
+import SentryDocumentTitle from 'sentry/components/sentryDocumentTitle';
 import TextCopyInput from 'sentry/components/textCopyInput';
 import apiApplication from 'sentry/data/forms/apiApplication';
 import {t} from 'sentry/locale';
 import ConfigStore from 'sentry/stores/configStore';
 import type {ApiApplication} from 'sentry/types/user';
 import getDynamicText from 'sentry/utils/getDynamicText';
-import DeprecatedAsyncView from 'sentry/views/deprecatedAsyncView';
+import {useApiQuery, useMutation, useQueryClient} from 'sentry/utils/queryClient';
+import useApi from 'sentry/utils/useApi';
+import {useParams} from 'sentry/utils/useParams';
 import SettingsPageHeader from 'sentry/views/settings/components/settingsPageHeader';
 
-type Props = RouteComponentProps<{appId: string}, {}>;
-type State = {
-  app: ApiApplication;
-} & DeprecatedAsyncView['state'];
-
-class ApiApplicationsDetails extends DeprecatedAsyncView<Props, State> {
-  rotateClientSecret = async () => {
-    try {
-      const rotateResponse = await this.api.requestPromise(
-        `/api-applications/${this.props.params.appId}/rotate-secret/`,
-        {
-          method: 'POST',
-        }
-      );
-      openModal(({Body, Header}) => (
-        <Fragment>
-          <Header>{t('Your new Client Secret')}</Header>
-          <Body>
-            <Alert type="info" showIcon>
-              {t('This will be the only time your client secret is visible!')}
-            </Alert>
-            <TextCopyInput aria-label={t('new-client-secret')}>
-              {rotateResponse.clientSecret}
-            </TextCopyInput>
-          </Body>
-        </Fragment>
-      ));
-    } catch {
-      addErrorMessage(t('Error rotating secret'));
+const PAGE_TITLE = t('Application Details');
+
+function getAppQueryKey(appId: string) {
+  return [`/api-applications/${appId}/`] as const;
```

Comment:
there's an ApiQueryKey you can use here instead of as const which helps with the 2nd argument as well
```suggestion
function getAppQueryKey(appId: string): ApiQueryKey {
  return [`/api-applications/${appId}/`];
```

Reply:
good to know, thx!